test/parallel/test-timers-clearImmediate.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+const N = 3;
+var count = 0;
+function next() {
+  const immediate = setImmediate(function() {
+    clearImmediate(immediate);
+    ++count;
+  });
+}
+for (var i = 0; i < N; ++i)
+  next();
+
+process.on('exit', () => {
+  assert.strictEqual(count, N, `Expected ${N} immediate callback executions`);
```

Comment:
@Fishrock123 Is this ok with you, since you previously requested a custom assertion message be added?